### DISCUSSION TITLE
How to add entity manager in resolver

### DISCUSSION THREAD

**Discussion by ace90** [overblog/GraphQLBundle#1003] [General]
| Q                | A
| ---------------- | -----
| Bug report?      | yes/no
| Feature request? | yes/no
| BC Break report? | yes/no
| RFC?             | yes/no
| Version/Branch   | x.y.z

I'm new in graphql. I try to add entity manger in my resolver using injecting service in constructor
```
class Characters
{
	public $em;

	public function __construct(EntityManagerInterface $em)
	{
		$this->em = $em;
	}

      public function getHumans()
          {
              dd($this);
              $humans = self::findByType(self::TYPE_HUMAN);
      
              return $humans;
          }
}
```

but when i query this. I always get this error `Using $this when not in object context`. May i know how can I fix this issue?

![Screenshot 2022-03-22 at 3 11 09 PM](https://user-images.githubusercontent.com/1217076/159427529-a49839ba-986c-469e-b87b-6b0072899450.png)

in my service.yml. I try this one too

```
App\Resolver\Characters:
        arguments: ['@doctrine.orm.entity_manager']
```

**Comment by mcg-web** [comment]
You should declare your service as a resolver. You can see how this can be done [here](https://github.com/overblog/GraphQLBundle/blob/master/docs/definitions/resolver.md).
